### PR TITLE
Add preferred anti-affinity

### PR DESCRIPTION
**Pull request by Hi-Fi**
Try to keep deployment pods at different nodes which helps with HA

### CODE REVIEW

File: .github/workflows/on_push.yml [hunk 1]
```diff
@@ -24,7 +24,7 @@ jobs:
         with:
           registry: ${{ env.CONTAINER_REGISTRY }}
           username: ${{ env.CONTAINER_REGISTRY_USER }}
-          password: ${{ secrets.CONTAINER_REGISTRY_PASSWORD }}
+          password: ${{ env.CONTAINER_REGISTRY_PASSWORD }}
```
Comment (frwickst) [inline .github/workflows/on_push.yml:27]: I don't know, to me this seems like a mighty bad idea? What was the reason for using env over secret here?

Comment (Hi-Fi) [reply .github/workflows/on_push.yml:27]: Duplication. As `CONTAINER_REGISTRY_PASSWORD` is stored to env at the beginning (https://github.com/andersinno/kolga/pull/9/files#diff-53dbc06610f897f66670a951aca3d54295c4a0fa85d7869295330aa05265a761R9), I don't see point of duplicating it's reading from secret. 

With the change it's also similar to registry and username variables.

**Review by joonvena — APPROVED** [review]
LGTM

File: helm/templates/03-deployment.yaml [hunk 2]
```diff
@@ -140,4 +140,27 @@ spec:
...
             claimName: {{ template "appname" . }}-pvc
       {{- end }}
+      {{ if or (.Values.replicaCount > 1) .Values.hpa.enabled }}
+      affinity:
+        podAntiAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+          - podAffinityTerm:
+              labelSelector:
+                matchLabels:
+                  name: app
```
Comment (antti-mikael) [inline helm/templates/03-deployment.yaml:150]: Correct my if I'm wrong but wouldn't this label selector match on pods that have a label `name=app`.

I think that what we actually want is more like:
```suggestion
                  app: {{ template "appname". }}
```


We could add the two other labels, `release` and `track`, as well.
```suggestion
                  app: "{{ template "appname". }}"
                  release: "{{ .Release.Name }}"
                  track: "{{ .Values.Application.track }}"
```

File: helm/templates/03-deployment.yaml [hunk 3]
```diff
@@ -140,4 +140,27 @@ spec:
...
+              topologyKey: kubernetes.io/hostname
+            weight: 100
+          - podAffinityTerm:
+              labelSelector:
+                matchLabels:
+                  name: app
```
Comment (antti-mikael) [inline helm/templates/03-deployment.yaml:156]: See the comment on line 150.

File: helm/templates/03-deployment.yaml [hunk 4]
```diff
@@ -140,4 +140,27 @@ spec:
...
+              topologyKey: topology.kubernetes.io/zone
+            weight: 100
+          - podAffinityTerm:
+              labelSelector:
+                matchLabels:
+                  name: app
```
Comment (antti-mikael) [inline helm/templates/03-deployment.yaml:162]: See the comment on line 150.

**Review by antti-mikael — CHANGES_REQUESTED** [review]

**Pull request merged**